frappe/public/js/frappe/views/reports/reportview.js

```diff
@@ -616,6 +617,14 @@ frappe.views.ReportView = frappe.ui.BaseList.extend({
 			me.render_view();
 		});
 	},
+	
+	make_check_all_button: function() {
+                var me = this;
+
+                this.page.add_inner_button(__('Un/Check All'), function() {
```

Comment:
The text of the button should change based on context. Initially it will be **Check All**, if atleast 1 row is checked, it should say **Uncheck All**

Reply:
@netchampfaris Please note, that the button text would be not correct, if it would change by the context. Please have a look at picture 4. and 5. The selection is always inversed, for example, if only some checkboxes are checked or unchecked.